runelite-client/src/main/java/net/runelite/client/plugins/prayer/PrayerCounter.java

```diff
@@ -22,32 +22,44 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package net.runelite.client.plugins.prayflick;
+package net.runelite.client.plugins.prayer;
 
-import com.google.common.eventbus.Subscribe;
-import javax.inject.Inject;
-import net.runelite.api.events.GameTick;
+import java.awt.image.BufferedImage;
+import java.util.function.Consumer;
+import lombok.Getter;
 import net.runelite.client.plugins.Plugin;
-import net.runelite.client.plugins.PluginDescriptor;
-import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.infobox.Counter;
 
-@PluginDescriptor(
-	name = "Prayer Flicking"
-)
-public class PrayerFlickPlugin extends Plugin
+public class PrayerCounter extends Counter
 {
-	@Inject
-	private PrayerFlickOverlay overlay;
+	@Getter
+	private PrayerType prayerType;
+	private BufferedImage image;
+	private static String text = "";
+
+	Consumer<BufferedImage> imageConsumer = (image) -> this.image = image;
```

Comment:
Why is this here? just create setter for image and reference that method.

Reply:
To get the image async from the cache. Eg. at loading up the game it won't retrieve the image.

Reply:
Yes. `PrayerCounter::setImage` == `Consumer<BufferedImage>`

Reply:
You learn something new everday, thanks ;)